simulator/src/main/java/com/hedera/block/simulator/mode/ConsumerModeHandler.java

```diff
@@ -34,31 +36,58 @@
  * {@link UnsupportedOperationException}.
  */
 public class ConsumerModeHandler implements SimulatorModeHandler {
+    /** Logger for this class */
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
 
+    // Configuration
     private final BlockStreamConfig blockStreamConfig;
 
+    // Service dependencies
+    private final ConsumerStreamGrpcClient consumerStreamGrpcClient;
+
     /**
-     * Constructs a new {@code ConsumerModeHandler} with the specified block stream configuration.
+     * Constructs a new {@code ConsumerModeHandler} with the specified dependencies.
      *
-     * @param blockStreamConfig the configuration data for managing block streams
+     * @param blockStreamConfig The configuration for block streaming parameters
+     * @param consumerStreamGrpcClient The client for consuming blocks via gRPC
+     * @throws NullPointerException if any parameter is null
      */
-    public ConsumerModeHandler(@NonNull final BlockStreamConfig blockStreamConfig) {
+    public ConsumerModeHandler(
+            @NonNull final BlockStreamConfig blockStreamConfig,
+            @NonNull final ConsumerStreamGrpcClient consumerStreamGrpcClient) {
         this.blockStreamConfig = requireNonNull(blockStreamConfig);
+        this.consumerStreamGrpcClient = requireNonNull(consumerStreamGrpcClient);
+    }
+
+    /**
+     * Initializes the gRPC channel for block consumption.
+     */
+    @Override
+    public void init() {
+        consumerStreamGrpcClient.init();
+        LOGGER.log(INFO, "gRPC Channel initialized for consuming blocks.");
     }
 
     /**
-     * Starts the simulator and initiate streaming, depending on the working mode.
+     * Starts consuming blocks from the stream beginning at genesis (block 0).
+     * Currently, requests an infinite stream of blocks starting from genesis.
+     *
+     * @throws InterruptedException if the consumption process is interrupted
      */
     @Override
-    public void start() {
-        throw new UnsupportedOperationException();
+    public void start() throws InterruptedException {
+        LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator is starting in consumer mode.");
+        consumerStreamGrpcClient.requestBlocks(0, 0);
     }
 
     /**
-     * Stops the handler and manager from streaming.
+     * Gracefully stops block consumption and shuts down the gRPC client.
+     *
+     * @throws InterruptedException if the shutdown process is interrupted
      */
     @Override
-    public void stop() {
-        throw new UnsupportedOperationException();
+    public void stop() throws InterruptedException {
+        consumerStreamGrpcClient.completeStreaming();
+        consumerStreamGrpcClient.shutdown();
```

Comment:
Generally it is a good idea to have a single 'teardown' type method because it is very error prone when when we need to call multiple things to shutdown something. It is a need to know solution and when used by someone else that someone else could not use the teardown logic properly as it was intended by the creator. I would suggest to streamline this, as well as in the other implementations of the handlers and clients in order to be clear how to shutdown (via interface) and then in the single shutdown method we should have all cleanup logic based on the specific implementation.

Reply:
I've streamlined it here.